crates/biome_js_analyze/src/lint/style/use_naming_convention.rs

```diff
@@ -386,20 +428,115 @@ declare_rule! {
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["private"]
+    ///                 },
+    ///                 "match": "_(.+)",
+    ///                 "formats": ["camelCase"]
+    ///             }
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
     /// If `match` is set and `formats` is unset,
-    /// then the part of the name captured by the regular expression is forwarded to the next convention of the array.
+    /// then the part of the name captured by the regular expression is forwarded to the next conventions of the array.
+    /// In the following example, we require that private class members start with `_` and all class members are in ["camelCase"].
+    ///
+    /// ```json5
+    /// {
+    ///     // ...
+    ///     "options": {
+    ///         "conventions": [
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["private"]
+    ///                 },
+    ///                 "match": "_(.+)"
+    ///                 // We don't need to specify `formats` because the capture is forwarded to the next conventions.
+    ///             },
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember"
+    ///                 },
+    ///                 "formats": ["camelCase"]
+    ///             }
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// If a declaration is not selected or if a capture is forwarded while there is no more conventions,
+    /// then the declaration name is verified against the default conventions.
+    /// Because, the default conventions already ensure that class members are in ["camelCase"],
+    /// the previous example can be simplified to:
+    ///
+    /// ```json5
+    /// {
+    ///     // ...
+    ///     "options": {
+    ///         "conventions": [
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["private"]
+    ///                 },
+    ///                 "match": "_(.+)"
+    ///                 // We don't need to specify `formats` because the capture is forwarded to the next conventions.
+    ///             }
+    ///             // default conventions
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// If the capture is identical to the initial name (it is not a part of the initial name),
+    /// then leading and trailing underscore and dollar signs are trimmed before checking against default conventions.
+    /// In the previous example, the capture is a part of the name because `_` is not included in the capture.
+    ///
+    /// You can reset all default conventions by adding a convention at the end of the array that accepts anything:
```

Comment:
```suggestion
    /// If the capture is identical to the initial name (it is not a part of the initial name),
    /// then, leading and trailing underscore and dollar signs are trimmed before being checked against default conventions.
    /// In the previous example, the capture is a part of the name because `_` is not included in the capture.
    ///
    /// You can reset all default conventions by adding a convention at the end of the array that accepts anything:
```